include/envoy/server/filter_config.h

```diff
@@ -58,6 +59,11 @@ class FactoryContext {
   virtual bool healthCheckFailed() PURE;
 
   /**
+   * @return HttpRouteManager& singleton for use by the entire server.
+   */
+  virtual HttpRouteManager& httpRouteManager() PURE;
```

Comment:
there is no reason to expose list() at the context level. Just have two interfaces, with one deriving from the other, and expose the first from the context.